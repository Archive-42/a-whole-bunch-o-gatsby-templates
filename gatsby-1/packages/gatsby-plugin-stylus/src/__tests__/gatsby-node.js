describe(`gatsby-plugin-stylus`, () => {
  jest.mock(`../resolve`, () => module => `/resolved/path/${module}`)

  const actions = {
    setWebpackConfig: jest.fn(),
  }

  // loaders "mocks"
  const loaders = {
    miniCssExtract: () => `miniCssExtract`,
    css: args => `css(${JSON.stringify(args)})`,
    postcss: args => `postcss(${JSON.stringify(args)})`,
    null: () => `null`,
  }

  const { onCreateWebpackConfig } = require(`../gatsby-node`)

  beforeEach(() => {
    actions.setWebpackConfig.mockReset()
  })

  const stylusPlugin = jest.fn().mockReturnValue(`foo`)

  const tests = {
    stages: [`develop`, `build-javascript`, `develop-html`, `build-html`],
    options: {
      "No options": {},
      "Stylus options #1": {
        use: [stylusPlugin()],
      },
      "Stylus options #2": {
        use: [stylusPlugin()],
        import: [`file.js`, `file2.js`],
      },
      "PostCss plugins": {
        postCssPlugins: [`test1`],
      },
      "css-loader use commonjs": {
        cssLoaderOptions: {
          esModule: false,
          modules: {
            namedExport: false,
          },
        },
      },
    },
  }

  tests.stages.forEach(stage => {
    for (const label in tests.options) {
      const options = tests.options[label]
      it(`Stage: ${stage} / ${label}`, () => {
        onCreateWebpackConfig(
          {
            actions,
            loaders,
            stage,
          },
          options
        )
        expect(actions.setWebpackConfig).toMatchSnapshot()
      })
    }
  })
})
