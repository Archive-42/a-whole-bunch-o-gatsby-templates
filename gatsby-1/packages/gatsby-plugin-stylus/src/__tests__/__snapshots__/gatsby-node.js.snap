// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-plugin-stylus Stage: build-html / No options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.styl\\$/,
                  "use": Array [
                    "css({\\"importLoaders\\":1,\\"modules\\":true})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/stylus-loader",
                      "options": Object {},
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.styl\\$/,
                  "use": Array [
                    "null",
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-html / PostCss plugins 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.styl\\$/,
                  "use": Array [
                    "css({\\"importLoaders\\":1,\\"modules\\":true})",
                    "postcss({\\"plugins\\":[\\"test1\\"]})",
                    Object {
                      "loader": "/resolved/path/stylus-loader",
                      "options": Object {},
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.styl\\$/,
                  "use": Array [
                    "null",
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-html / Stylus options #1 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.styl\\$/,
                  "use": Array [
                    "css({\\"importLoaders\\":1,\\"modules\\":true})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/stylus-loader",
                      "options": Object {
                        "use": Array [
                          "foo",
                        ],
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.styl\\$/,
                  "use": Array [
                    "null",
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-html / Stylus options #2 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.styl\\$/,
                  "use": Array [
                    "css({\\"importLoaders\\":1,\\"modules\\":true})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/stylus-loader",
                      "options": Object {
                        "import": Array [
                          "file.js",
                          "file2.js",
                        ],
                        "use": Array [
                          "foo",
                        ],
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.styl\\$/,
                  "use": Array [
                    "null",
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-html / css-loader use commonjs 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.styl\\$/,
                  "use": Array [
                    "css({\\"esModule\\":false,\\"modules\\":{\\"namedExport\\":false},\\"importLoaders\\":1})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/stylus-loader",
                      "options": Object {},
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.styl\\$/,
                  "use": Array [
                    "null",
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-javascript / No options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.styl\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"importLoaders\\":1,\\"modules\\":true})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/stylus-loader",
                      "options": Object {},
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.styl\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"importLoaders\\":2,\\"modules\\":false})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/stylus-loader",
                      "options": Object {},
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-javascript / PostCss plugins 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.styl\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"importLoaders\\":1,\\"modules\\":true})",
                    "postcss({\\"plugins\\":[\\"test1\\"]})",
                    Object {
                      "loader": "/resolved/path/stylus-loader",
                      "options": Object {},
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.styl\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"importLoaders\\":2,\\"modules\\":false})",
                    "postcss({\\"plugins\\":[\\"test1\\"]})",
                    Object {
                      "loader": "/resolved/path/stylus-loader",
                      "options": Object {},
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-javascript / Stylus options #1 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.styl\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"importLoaders\\":1,\\"modules\\":true})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/stylus-loader",
                      "options": Object {
                        "use": Array [
                          "foo",
                        ],
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.styl\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"importLoaders\\":2,\\"modules\\":false})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/stylus-loader",
                      "options": Object {
                        "use": Array [
                          "foo",
                        ],
                      },
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-javascript / Stylus options #2 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.styl\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"importLoaders\\":1,\\"modules\\":true})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/stylus-loader",
                      "options": Object {
                        "import": Array [
                          "file.js",
                          "file2.js",
                        ],
                        "use": Array [
                          "foo",
                        ],
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.styl\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"importLoaders\\":2,\\"modules\\":false})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/stylus-loader",
                      "options": Object {
                        "import": Array [
                          "file.js",
                          "file2.js",
                        ],
                        "use": Array [
                          "foo",
                        ],
                      },
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: build-javascript / css-loader use commonjs 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.styl\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"esModule\\":false,\\"modules\\":{\\"namedExport\\":false},\\"importLoaders\\":1})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/stylus-loader",
                      "options": Object {},
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.styl\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"importLoaders\\":2,\\"esModule\\":false,\\"modules\\":false})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/stylus-loader",
                      "options": Object {},
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop / No options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.styl\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"importLoaders\\":1,\\"modules\\":true})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/stylus-loader",
                      "options": Object {},
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.styl\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"importLoaders\\":2,\\"modules\\":false})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/stylus-loader",
                      "options": Object {},
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop / PostCss plugins 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.styl\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"importLoaders\\":1,\\"modules\\":true})",
                    "postcss({\\"plugins\\":[\\"test1\\"]})",
                    Object {
                      "loader": "/resolved/path/stylus-loader",
                      "options": Object {},
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.styl\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"importLoaders\\":2,\\"modules\\":false})",
                    "postcss({\\"plugins\\":[\\"test1\\"]})",
                    Object {
                      "loader": "/resolved/path/stylus-loader",
                      "options": Object {},
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop / Stylus options #1 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.styl\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"importLoaders\\":1,\\"modules\\":true})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/stylus-loader",
                      "options": Object {
                        "use": Array [
                          "foo",
                        ],
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.styl\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"importLoaders\\":2,\\"modules\\":false})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/stylus-loader",
                      "options": Object {
                        "use": Array [
                          "foo",
                        ],
                      },
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop / Stylus options #2 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.styl\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"importLoaders\\":1,\\"modules\\":true})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/stylus-loader",
                      "options": Object {
                        "import": Array [
                          "file.js",
                          "file2.js",
                        ],
                        "use": Array [
                          "foo",
                        ],
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.styl\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"importLoaders\\":2,\\"modules\\":false})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/stylus-loader",
                      "options": Object {
                        "import": Array [
                          "file.js",
                          "file2.js",
                        ],
                        "use": Array [
                          "foo",
                        ],
                      },
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop / css-loader use commonjs 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.styl\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"esModule\\":false,\\"modules\\":{\\"namedExport\\":false},\\"importLoaders\\":1})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/stylus-loader",
                      "options": Object {},
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.styl\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css({\\"importLoaders\\":2,\\"esModule\\":false,\\"modules\\":false})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/stylus-loader",
                      "options": Object {},
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop-html / No options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.styl\\$/,
                  "use": Array [
                    "css({\\"importLoaders\\":1,\\"modules\\":true})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/stylus-loader",
                      "options": Object {},
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.styl\\$/,
                  "use": Array [
                    "null",
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop-html / PostCss plugins 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.styl\\$/,
                  "use": Array [
                    "css({\\"importLoaders\\":1,\\"modules\\":true})",
                    "postcss({\\"plugins\\":[\\"test1\\"]})",
                    Object {
                      "loader": "/resolved/path/stylus-loader",
                      "options": Object {},
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.styl\\$/,
                  "use": Array [
                    "null",
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop-html / Stylus options #1 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.styl\\$/,
                  "use": Array [
                    "css({\\"importLoaders\\":1,\\"modules\\":true})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/stylus-loader",
                      "options": Object {
                        "use": Array [
                          "foo",
                        ],
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.styl\\$/,
                  "use": Array [
                    "null",
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop-html / Stylus options #2 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.styl\\$/,
                  "use": Array [
                    "css({\\"importLoaders\\":1,\\"modules\\":true})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/stylus-loader",
                      "options": Object {
                        "import": Array [
                          "file.js",
                          "file2.js",
                        ],
                        "use": Array [
                          "foo",
                        ],
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.styl\\$/,
                  "use": Array [
                    "null",
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`gatsby-plugin-stylus Stage: develop-html / css-loader use commonjs 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.styl\\$/,
                  "use": Array [
                    "css({\\"esModule\\":false,\\"modules\\":{\\"namedExport\\":false},\\"importLoaders\\":1})",
                    "postcss({})",
                    Object {
                      "loader": "/resolved/path/stylus-loader",
                      "options": Object {},
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.styl\\$/,
                  "use": Array [
                    "null",
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
